FerramAerospaceResearch/FARAeroComponents/VehicleOcclusion.cs

```diff
@@ -0,0 +1,828 @@
+        private readonly SphereDistanceInfo[] convectionPoints = new SphereDistanceInfo[3];
+        private readonly SphereDistanceInfo[] sunPoints = new SphereDistanceInfo[3];
+        private readonly SphereDistanceInfo[] bodyPoints = new SphereDistanceInfo[3];
+        
+        private readonly DirectionPreprocessInfo[] directionPreprocessInfos = new DirectionPreprocessInfo[3];
+        private readonly List<Vector3> directionList = new List<Vector3>(OcclusionSettings.MaxJobs);
+        private readonly List<SphereDistanceInfo[]> occlusionPointsList = new List<SphereDistanceInfo[]>(3);
+        public float3 DefaultAngleWeights => new float3(0.85f, 0, 0.15f);
+
+        // VehicleVoxel center and extents
+        private Vector3 center = Vector3.zero;
+        private Vector3 extents = new Vector3(10, 10, 10);
+
+        private int jobsInProgress;
+        private float averageMissOnBestAngle = 0;
+
+        // Large long-term allocations, memory re-used in each
+        // Cleaned and Allocated in ResetCalculations().  Disposed in OnDestroy()
+        private NativeMultiHashMap<int, int> indexedPriorityMap;
+
+        private readonly NativeArray<RaycastCommand>[] allCasts = new NativeArray<RaycastCommand>[OcclusionSettings.MaxJobs];
+        private readonly NativeArray<RaycastHit>[] allHits = new NativeArray<RaycastHit>[OcclusionSettings.MaxJobs];
+        private readonly NativeMultiHashMap<int, int>[] allHitsMaps = new NativeMultiHashMap<int, int>[OcclusionSettings.MaxJobs];
+        private readonly NativeHashMap<int, float>[] allHitSizeMaps = new NativeHashMap<int, float>[OcclusionSettings.MaxJobs];
+        private readonly NativeArray<float>[] allSummedHitAreas = new NativeArray<float>[OcclusionSettings.MaxJobs];
+
+        // allCasts => allHits
+        // allHits => allHitMaps
+        // allHitMaps => allHitSizeMaps,
+        //               allSummedHitAreas,
+
+        private readonly RaycastJobInfo[] raycastJobTracker = new RaycastJobInfo[OcclusionSettings.MaxJobs];
+        public bool RequestReset = false;
+        public Utils.Metrics metrics = new Utils.Metrics();
+        public const string FixedUpdateMetric = "FixedUpdateStall";
+        public const string UpdateMetric = "UpdateStall";
+        private System.Diagnostics.Stopwatch JobCompleteWatch = new System.Diagnostics.Stopwatch();
+
+        public void Setup(FARVesselAero farVesselAero)
+        {
+            this.farVesselAero = farVesselAero;
+            Vessel = farVesselAero.Vessel;
+        }
+
+        //Extents = axis-aligned bounding box dimensions of combined vehicle meshes
+        public void SetVehicleBounds(Vector3d center, Vector3d extents)
+        {
+            this.center = center;
+            this.extents = extents;
+            FARLogger.Info($"[VehicleOcclusion] {Vessel?.name} Learned Center {center} and Extents {extents}");
+        }
+
+        private void DisposeLongTermAllocations()
```

Comment:
Implement `IDisposable` instead

Reply:
I don't understand what to do here, as the scope for these is not something I can place into a try/finally, or into a using statement as I understand it.

Reply:
It's just changing the method name to `Dispose`, scope doesn't matter but `IDisposable` shows that the class has resources that need to be cleaned up